examples_tests/01.HelloWorld/main.cpp

```diff
@@ -21,39 +28,282 @@ inline void debugCallback(nbl::video::E_DEBUG_MESSAGE_SEVERITY severity, nbl::vi
 	const char* sev = nullptr;
 	switch (severity)
 	{
-	case video::EDMS_VERBOSE:
+        case video::EDMS_VERBOSE:
 		sev = "verbose"; break;
-	case video::EDMS_INFO:
+        case video::EDMS_INFO:
 		sev = "info"; break;
-	case video::EDMS_WARNING:
+        case video::EDMS_WARNING:
 		sev = "warning"; break;
-	case video::EDMS_ERROR:
+        case video::EDMS_ERROR:
 		sev = "error"; break;
 	}
 	std::cout << "OpenGL " << sev << ": " << msg << std::endl;
 }
 
+// Don't wanna use Printer::log
+#define LOG(...) printf(__VA_ARGS__); printf("\n");
+class DemoEventCallback : public nbl::ui::IWindow::IEventCallback
+{
+	void onWindowShown_impl() override
+	{
+		LOG("Window Shown");
+	}
+	void onWindowHidden_impl() override
+	{
+		LOG("Window hidden");
+	}
+	void onWindowMoved_impl(int32_t x, int32_t y) override
+	{
+		LOG("Window window moved to { %d, %d }", x, y);
+	}
+	void onWindowResized_impl(uint32_t w, uint32_t h) override
+	{
+		LOG("Window resized to { %u, %u }", w, h);
+	}
+	void onWindowMinimized_impl() override
+	{
+		LOG("Window minimized");
+	}
+	void onWindowMaximized_impl() override
+	{
+		LOG("Window maximized");
+	}
+	void onGainedMouseFocus_impl() override
+	{
+		LOG("Window gained mouse focus");
+	}
+	void onLostMouseFocus_impl() override
+	{
+		LOG("Window lost mouse focus");
+	}
+	void onGainedKeyboardFocus_impl() override
+	{
+		LOG("Window gained keyboard focus");
+	}
+	void onLostKeyboardFocus_impl() override
+	{
+		LOG("Window lost keyboard focus");
+	}
+
+	void onMouseConnected_impl(core::smart_refctd_ptr<nbl::ui::IMouseEventChannel>&& mch) override
+	{
+		LOG("A mouse has been connected");
+	}
+	void onMouseDisconnected_impl(nbl::ui::IMouseEventChannel* mch) override
+	{
+		LOG("A mouse has been disconnected");
+	}
+	void onKeyboardConnected_impl(core::smart_refctd_ptr<nbl::ui::IKeyboardEventChannel>&& kbch) override
+	{
+		LOG("A keyboard has been connected");
+	}
+	void onKeyboardDisconnected_impl(nbl::ui::IKeyboardEventChannel* mch) override
+	{
+		LOG("A keyboard has been disconnected");
+	}
+};
 
 int main()
 {
 	constexpr uint32_t WIN_W = 800u;
 	constexpr uint32_t WIN_H = 600u;
 	constexpr uint32_t SC_IMG_COUNT = 3u;
-
-	auto win = CWindowT::create(WIN_W, WIN_H, system::IWindow::ECF_NONE);
-
+	
+	// Note(achal): This is unused, for now
 	video::SDebugCallback dbgcb;
 	dbgcb.callback = &debugCallback;
 	dbgcb.userData = nullptr;
-	auto gl = video::IAPIConnection::create(video::EAT_OPENGL, 0, "New API Test", &dbgcb);
-	auto surface = gl->createSurface(win.get());
+	
+	auto system = CommonAPI::createSystem();

```

Comment:
Example 01 should refrain from using `CommonAPI` (everything afterwards is fine to do so)

Reply:
noted